### PR TITLE
Refactor Releaf::Settings registration and normalization

### PR DESCRIPTION
**Pull request by miks**

### CODE REVIEW

File: releaf-core/app/lib/releaf/settings/normalize_value.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  def self.normalize_float(value)
+    value.to_s.sub(",", ".").to_f
+  end
+
+  def self.normalize_integer(value)
+    value.to_i
+  end
+
+  def self.normalize_time(value)
+    Time.parse(value)
```
Comment (graudeejs) [inline releaf-core/app/lib/releaf/settings/normalize_value.rb]:
```
Time.parse('')
TIme.parse(nil)
Time.parse('foo')
Time.parse(1)
...
```
all will fail with some sort of exception (ArgumentError or TypeError).

Is that intentional?
Same for DateTime and Date.

**Comment by miks** [comment]
We can't cover all cases. Also, validation will be overkill for such a simple model.
Please note that field will be rendered with `releaf_time_field`.
What's your suggestion?

**Comment by graudeejs** [comment]
Perhaps:

```ruby
return nil if value.blank?
```

That should be acceptable (for optional fields at least)

**Comment by miks** [comment]
Fixed with 3914e7c

**Pull request merged**